frontend/src/components/ModelInputs.tsx

```diff
@@ -0,0 +1,127 @@
+
+const DEFAULTS = {
+    O2: 30,
+    H2O: 30,
+    H2S: 0,
+    SO2: 10,
+    NO2: 20,
+};
+
+const PPM_MAX = 1000000;
+
+function SubstanceAdder({ invisible, onAdd }: { invisible: string[]; onAdd: (subst: string) => void }) {
+    const [selected, setSelected] = useState<string | null>(null);
+
+    if (invisible.length === 0) return;
+    return (
+        <div style={{ display: "flex", alignItems: "center", marginTop: "1em" }}>
+            <Autocomplete
+                style={{ flexGrow: 1 }}
+                id="newConcentration"
+                label=""
+                placeholder="Add new"
+                options={invisible}
+                onOptionsChange={({ selectedItems }) => setSelected(selectedItems[0])}
+                selectedOptions={selected === null ? [] : [selected]}
+            />
+            <Button
+                onClick={() => {
+                    onAdd(selected!);
+                    setSelected(null);
+                }}
+                disabled={selected === null}
+            >
+                +
+            </Button>
+        </div>
+    );
+}
+
+function ParametersInput({
+    model,
+    parameters,
+    setParameter,
+}: {
+    model: ModelConfig;
+    parameters: Record<string, number>;
+    setParameter: (name: string, value: number) => void;
+}) {
+    if (!model.parameters) return;
+    return (
+        <div style={{ paddingTop: "1em" }}>
+            <Typography bold> Input parameters </Typography>
+            {Object.entries(model.parameters).map(([name, config], index) => (
+                <TextField
+                    type="number"
+                    key={index}
+                    label={config.label}
+                    style={{ paddingTop: "5px" }}
+                    step="any"
+                    unit={config.unit ?? config.custom_unit}
+                    max={config.maximum}
+                    min={config.minimum}
+                    value={parameters[name]}
+                    onChange={(e: ChangeEvent<HTMLInputElement>) => setParameter(name, e.target.valueAsNumber)}
+                />
+            ))}
+        </div>
+    );
+}
+
+function getParameterDefaults(model: ModelConfig) {
+    return Object.fromEntries(Object.entries(model.parameters).map(([key, value]) => [key, value.default]));
+}
+
+const ModelInputs: React.FC<{
+    model: ModelConfig;
+    visible: boolean;
+    onSubmit: (concentrations: Record<string, number>, parameters: Record<string, number>) => void;
+}> = ({ model, visible: not_hidden, onSubmit }) => {
+    const [concentrations, setConcentrations] = useState<Record<string, number>>(DEFAULTS);
+    const [visible, setVisible] = useState<string[]>(Object.keys(DEFAULTS));
+    const [parameters, setParameters] = useState<Record<string, number>>(getParameterDefaults(model));
+
+    if (!not_hidden) {
+        return <></>;
+    }
+
+    const invisible = model.validSubstances.filter((subst) => !visible.includes(subst));
```

Comment:
this hidden vs invisible.. Let's rename this to `addableItems` or `itemsToAdd` or something similar.